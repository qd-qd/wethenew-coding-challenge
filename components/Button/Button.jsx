import PropTypes from 'prop-types';
import styles from './Button.module.css';

const Button = ({ label, onClick }) => (
  <button onClick={onClick} type="button" className={styles.button}>
    {label}
  </button>
);

Button.propTypes = {
  label: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,
};

export default Button;
