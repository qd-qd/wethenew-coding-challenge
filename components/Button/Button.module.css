.button {
  color: var(--theme-color-white);
  background-color: var(--theme-color-black);
  padding: 1rem 1.75rem;
  box-shadow: var(--theme-elevation-2);
  border: none;
  width: 100%;
  border-radius: var(--theme-shape-radius);
}

.button:hover {
  box-shadow: var(--theme-elevation-8);
  outline: 0;
}

.button::first-letter {
  text-transform: uppercase;
}

@media (min-width: 35em) {
  .button {
    max-width: 250px;
  }
}
