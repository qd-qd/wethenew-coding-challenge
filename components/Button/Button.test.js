import { render, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Button from './Button';

describe('Button', () => {
  test('displays greeting', () => {
    const handler = () => {};
    const label = 'button-test';

    render(<Button onClick={handler} label={label} />);
    expect(screen.getByText(label)).toBeDefined();
    expect(screen.getByText(label)).toHaveAttribute('class');
    expect(screen.getByText(label)).toHaveAttribute('type');
  });

  test('is clickable', () => {
    const handler = jest.fn(e => e.preventDefault());
    const label = 'button-test';

    render(<Button onClick={handler} label={label} />);
    fireEvent.click(screen.getByText(label), 'leftClick');
    expect(handler).toHaveBeenCalledTimes(1);
  });

  test("doesn't break snapshot", () => {
    const handler = () => {};
    const label = 'button-test';
    const { container } = render(<Button onClick={handler} label={label} />);
    expect(container).toMatchSnapshot();
  });
});
