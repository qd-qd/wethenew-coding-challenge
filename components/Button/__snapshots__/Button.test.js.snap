// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button doesn't break snapshot 1`] = `
<div>
  <button
    class="button"
    type="button"
  >
    button-test
  </button>
</div>
`;
